pyteal/ast/router.py

```diff
@@ -877,6 +878,24 @@ def build_program(
             * clear_state_program: an AST for clear-state program
             * contract: a Python SDK Contract object to allow clients to make off-chain calls
         """
+        if self.bare_calls and not self.bare_calls.is_empty():
```

Comment:
Was also considering of privatizing `build_program()`. What do you think?

Reply:
yes, I up the vote for privatizing `build_program`, for I hope *for now* folks are using only generated code from router, rather than the composed Expr result...

Reply:
and if we go down the path of rewinding the scratch slot id, we should not expect people to continue compiling in the same session of compiling such router program, or they will end up using same scratch slots, which will cause trouble...

Reply:
Yes, let's chat about it.

Reply:
Resolving as discussed